ws-server/src/tests.rs

```diff
@@ -126,24 +109,22 @@ async fn invalid_request_object() {
 
 #[tokio::test]
 async fn register_methods_works() {
-	let server = WsServer::new("127.0.0.1:0").await.unwrap();
-	assert!(server.register_method("say_hello".to_owned()).is_ok());
-	assert!(server.register_method("say_hello".to_owned()).is_err());
-	assert!(server.register_notification("notif".to_owned(), false).is_ok());
-	assert!(server.register_notification("notif".to_owned(), false).is_err());
-	assert!(server.register_subscription("subscribe_hello".to_owned(), "unsubscribe_hello".to_owned()).is_ok());
-	assert!(server.register_subscription("subscribe_hello_again".to_owned(), "notif".to_owned()).is_err());
+	let mut server = WsServer::new("127.0.0.1:0").await.unwrap();
+	assert!(server.register_method("say_hello", |_| Ok("lo")).is_ok());
+	assert!(server.register_method("say_hello", |_| Ok("lo")).is_err());
+	assert!(server.register_subscription("subscribe_hello", "unsubscribe_hello").is_ok());
+	assert!(server.register_subscription("subscribe_hello_again", "unsubscribe_hello").is_err());
```

Comment:
Do you think it is actually useful to users to be able to provide the unsubscribe method name like this? It seems like a pretty niche feature and I fail to see why it's useful. Can we get rid of it and just prepend `un` to the subscription method name?

Reply:
I defaulted on not changing the API too much here, but I also reckon explicit names are better than implicit conventions.